Makefile

```
@@ -35,11 +35,25 @@ UI_WEBVIEW_LDFLAGS=-ldflags "$(UI_LDFLAGS_CONTENT)$(if $(filter windows,$(GOOS))
-.PHONY: build wallet wallet-binary wallet-webview wallet-binary-webview webkit-shim bundle-macos pkg-macos pkg-macos-adhoc mod-tidy clean test
+.PHONY: build wallet wallet-binary wallet-webview wallet-binary-webview webui-embed webkit-shim bundle-macos pkg-macos pkg-macos-adhoc mod-tidy clean test
 
 # Alias for building program binary
 build: $(BINARIES)
 
+# Copy the built SPA (ui/web/dist, gitignored) into the //go:embed dist target
+# so the wallet binaries embed the real UI. Vite builds into ui/web/dist, so
+# `npm run build` never overwrites the tracked ui/internal/webui/dist/index.html
+# placeholder; the real bundle lands in the embed dir only here, at binary-build
+# time. No-ops (keeping the placeholder) when the web bundle has not been built.
+webui-embed:
```

checkmake (0.3.2)

[warning] Target body for "webui-embed" exceeds allowed length of 5 lines (7).

(maxbodylength)